src/ja/server/database/types/work_machine.py

```diff
@@ -57,6 +75,16 @@ def deallocate(self,
           False if the amount of resources to be freed exceeds the total amount
           of allocated resources.
         """
+        if (self._total_resources - (self._free_resources + allocation)).is_negative():
+            return False
+
+        self._free_resources += allocation
+        return True
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, WorkMachineResources):
+            return self._total_resources == other.total_resources and self._free_resources == other.free_resources
+        return False
 
 
 class WorkMachineConnectionDetails:
```

Comment:
What do we do with this class.

Reply:
It's a TODO, I am not sure what exactly we'll need.